test/unit/browser.test.ts

```diff
@@ -11,6 +11,7 @@ describe('browser client', function () {
     client = Singleton.factory({
       logger: nullLogger()
     });
+    client.clear()
```

Comment:
I suspect you had to do this because now there is a singleton of the `GlobalStore` which means it might hold the state of other `Honeybadger` clients right?
Perhaps, it would be better to revert to the original usage of the `GlobalStore` to make sure we have separate context for each `Honeybadger` client. What do you think?
For ALS, we opted for the singleton, because we get a separate context every time we call `run()`.

Reply:
True. But the inconsistency always was a bit jarring. However, a big reason why it's a singleton is because we need to be able to recover breadcrumbs from before we switched to async. ie this flow:

```
store is Global
do stuff that creates breadcrumbs
enter async context
switch store to Async
error happens in async context
report in async context (gather breadcrumbs and other event data) <-- a
something goes wrong while reporting
report new thing in global context (gather breadcrumbs and other event data) <--b
```

On line a: when gathering breadcrumbs to report, we also need breadcrumbs that happened before we switched to Async. That's what [this line](https://github.com/honeybadger-io/honeybadger-js/pull/845/files#diff-76ffc16908873b969ebaba241404e1554078b511be4fc2bd8a1f63b044041e98R440) does. But now that I think of it, we don't really need a singleton for that. We can do it earlier, by initialising the ALS store with the global one:

```js
AsyncStore.run({ ...GlobalStore.getStore() }, () => {
  // code
});
```

But the second point is stage b: when we exit the async context, we lose access to the async store, so the only way to recover any breadcrumbs that happened before entering the async context is by having the global store be a singleton.

I think the singleton approach is fine, since the Honeybadger client is also a global singleton. But now that you mention it, I see that a user could manually create another client with `factory()`. I'll update that method to reset the context for the created client.